core/foundation/src/FoundationUtils.cxx

```diff
@@ -88,5 +89,89 @@ std::string MakePathRelative(const std::string &path, const std::string &base, b
    }
    return result;
 }
+
+/// Transforms a file path by replacing its backslashes with slashes.
+void ConvertToUnixPath(std::string& Path) {
+   std::replace(Path.begin(), Path.end(), '\\', '/');
+}
+
+const char* GetFallbackRootSys() {
+#ifdef WIN32
+   static char lpFilename[_MAX_PATH];
+   if (::GetModuleFileNameA(
+          NULL,                   // handle to module to find filename for
+          lpFilename,             // pointer to buffer to receive module path
+          sizeof(lpFilename))) {  // size of buffer, in characters
+      auto parent_path = [](std::string& path) {
+         return path.substr(0, path.find_last_of("/\\"));
+      };
+      return StrDup(parent_path(parent_path(lpFilename)).c_str());
+   }
+   return nullptr;
+#else
+   return "/usr/local/root";
+#endif
+}
+
+const std::string& GetRootSys() {
+#ifdef ROOTPREFIX
+   if (IgnorePrefix()) {
+#endif
+      static std::string rootsys;
+      if (rootsys.empty()) {
+         rootsys = ::getenv("ROOTSYS");
+         // We cannot use gSystem->UnixPathName.
+         ConvertToUnixPath(rootsys);
+      }
+      // FIXME: Should this also call UnixPathName for consistency?
+      if (rootsys.empty())
+         rootsys = GetFallbackRootSys();
+      return rootsys;
+#ifdef ROOTPREFIX
+   } else {
+      const static std::string rootsys = ROOTPREFIX;
+      return rootsys;
+   }
+#endif
+}
+
+
+const std::string& GetIncludeDir() {
+#ifdef ROOTINCDIR
+   if (IgnorePrefix()) {
+#endif
+      static std::string rootincdir;
+      if (rootincdir.empty()) {
+         const std::string& sep = GetPathSeparator();
+         rootincdir = GetRootSys() + sep + "include" + sep;
```

Comment:
These hard-coded values are the reason why we have [ROOT-9569](https://sft.its.cern.ch/jira/browse/ROOT-9569). We should use what was specified at configuration time, i.e. `${CMAKE_INSTALL_INCLUDEDIR}`. The include dir is "include/root" when `gnuinstall=ON`, for example.

Reply:
If the `CMAKE_INSTALL_INCLUDEDIR` is set it sets `ROOTINCDIR` and we return the correct value AFAICT.

Reply:
That's only true when gnuinstall=ON, and it's usually OFF. See https://github.com/root-project/root/blob/13b66e239ccd1bd53cc1acb8405b32fb7610eaa9/config/RConfigure.in#L8-L21.

Reply:
I see, although, I am not sure what would be the right fix for this. Should we set `ROOTINCDIR` independently on `gnuinstall=ON`?

Reply:
> Should we set ROOTINCDIR independently on gnuinstall=ON?

Yes, most likely that will be required to fix the issues we have for installation. See https://sft.its.cern.ch/jira/browse/ROOT-10151 for a full list of the FHS related issues.